### ISSUE TITLE
Not able to cite paper without authors 

### ISSUE THREAD

**Issue by cparsania** [nucleic-acid/namedropR#44]
Hi,

This is the paper (https://www.nature.com/articles/s41587-022-01346-7) I want to cite but it is not able to detect from .bib file. 

Thanks.

**Comment by nucleic-acid** [comment]
Hi, thanks for the report. I'd like to figure this out:

Do you get any output / warning / error message? When I obtain the bib for the reference it returns this:
![2022 html](https://user-images.githubusercontent.com/76264370/169465030-93f84adb-cadf-4830-b0f6-d4117aad200f.png)

So yes, the NA is not as it should be (I'll fix that ASAP), but I still do get an output. Could you please post the content of the bib file you use and also the function call, that causes the issue?

**Comment by cparsania** [comment]
Hi, 

This is the command line output 

```
namedropR::drop_name("path/to/export.bib")
Years coerced to string format.
Bibliography file successfully read.
No cite_key specified. Working through all possible 0 entries in the bibliography.
[1] "No Visual Citation created."
Warning message:
In namedropR::drop_name("/Users/chiragp/Downloads/export.bib") :
  No reference matches the given cite_keys. Please check that citation key(s) are correct.

```
text from export.bib:

```

@article{,
   abstract = {Two upcoming regulatory decisions represent a tipping point for commercial gene therapy, with implications for work on existing viral vectors and the pursuit of new ones.},
   doi = {10.1038/s41587-022-01346-7},
   issn = {1546-1696},
   issue = {5},
   journal = {Nature Biotechnology 2022 40:5},
   keywords = {Agriculture,Bioinformatics,Biomedical Engineering/Biotechnology,Biomedicine,Biotechnology,Life Sciences,general},
   month = {5},
   pages = {621-621},
   publisher = {Nature Publishing Group},
   title = {Gene therapy at the crossroads},
   volume = {40},
   url = {https://www.nature.com/articles/s41587-022-01346-7},
   year = {2022},
} 


```

**Comment by nucleic-acid** [comment]
Ah, I see whats going on. The reference from the exported bib file lacks a unique identifier (a "cite_key") that allows the function to differentiate the references. That's why it returns "No reference matches the given cite_keys. Please check that citation key(s) are correct." As no key is given it cannot select the reference and extract its data.

Some reference manager softwares like Zotero can automatically assign a unique key when exporting to bibtex. Some others might not do that, so I will incorporate a fix to do so. Thanks to catch this! I'll close this issue, once the update is in place.

A quick fix for you could be to give the reference above a unique key and then it should work. I usually use "author_year" like e.g. "Jones_2011" but it can be anything. It goes right after the opening curly braces. In the example you gave this could be: 

```
@article{nature_biotech_2022,
   abstract = {Two upcoming regulatory decisions represent a tipping point for commercial gene therapy, with implications for work on existing viral vectors and the pursuit of new ones.},
   doi = {10.1038/s41587-022-01346-7},
   issn = {1546-1696},
   issue = {5},
   journal = {Nature Biotechnology 2022 40:5},
   keywords = {Agriculture,Bioinformatics,Biomedical Engineering/Biotechnology,Biomedicine,Biotechnology,Life Sciences,general},
   month = {5},
   pages = {621-621},
   publisher = {Nature Publishing Group},
   title = {Gene therapy at the crossroads},
   volume = {40},
   url = {https://www.nature.com/articles/s41587-022-01346-7},
   year = {2022},
} 
```

If you have >1 reference in the file the keys need to be unique. Can you please try this and check if it works?

**Comment by nucleic-acid** [comment]
This should be fixed in the latest development version v2.3.3.9002. the paper you gave as example now returns this:
![GenetheraNatureBio2022 html](https://user-images.githubusercontent.com/76264370/170890981-6e1c7686-b925-4f7f-b4f0-57fd3a0a4b25.png)

If you're using the dev version, I recommend updating. This will come to the next CRAN release, but that might take some more days/weeks.
